student.rb

```diff
@@ -1,11 +1,18 @@
 require_relative 'person'
 
 class Student < Person
+  attr_reader :classroom
+
   def initialize(classroom, age, name: 'unknown', parent_permission: true)
     super(name, age, parent_permission)
     @classroom = classroom
   end
 
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.student.push(self) unless classroom.student.include?(self)
```

Comment:
A classroom should be added when a student is initialized without explicitly calling the classroom setter. Kindly move this code to the initialize method.